### DISCUSSION TITLE
How does events work when using RemoteTargets?

### DISCUSSION THREAD

**Discussion by roozbehid** [microsoft/vs-streamjsonrpc#1031] [General]
I have a middleman that has multiple connections to multiple clients, then this middleman has one connection to Server. I was assuming somehow notifications from server would work but it seems it is not. I am assuming jsonRPC in the middleman doesnt know how to route them(maybe?)

I am just confused how to make it work? For example assuming having 2 clients, I can see my `Subscribe(IObservable<int>)` being called two times (meaning `client1` and `client2` make calls and I see them in `server1`). But `server1` invoking those observables does not end up on all of clients (client1 and client2) to recieve notifications. And middleman is the one which used RemoteTargets, meaning `client1` and `client2` are connected to `middleman1`. `middleman1` is connected to `server1`.

**Comment by AArnott** [comment]
Good question. @tinaschrepfer may be able to answer your question as she contributed the RemoteTargets feature. It was added very particularly for LSP scenarios, so it _may_ not be as generalizable to other scenarios as the rest of the library.
